content/en/blog/2022/how-to-get-started-with-opentelemetry-go.md

```diff
@@ -0,0 +1,420 @@
+
+OpenTelemetry serves as a standard observability framework that captures all data under a single specification. It provides several components, including:
+
+1. APIs and SDKs per programming language for generating telemetry
+2. The OpenTelemetry Collector; receives, processes, and exports telemetry data to different destinations.
+3. [OTLP](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md) protocol for shipping telemetry data
+   To get a deeper understanding of this technology, including its structure and the primary motivation, visit this guide.
+
+For this OpenTelemetry Golang guide, here are the terms you need to know:
+
+- **Span:** Represents a single operation within a Trace.
+- **Trace:** [‘Call-stacks’ for distributed services](https://www.aspecto.io/blog/guide-to-distributed-tracing/). A DAG of Spans, where the edges between Spans are defined as parent/child. relationship.
+- **Exporter:** Provides functionality to emit telemetry to consumers. Used by Instrumentation Libraries and the Collector. Exporters can be push- or pull-based.
+- **Context propagation:** Allows all Data Sources to share an underlying context mechanism for storing state and accessing data across the lifespan of a Transaction. See context propagation spec.
+- **Instrumentation:** Denotes the Library that provides the instrumentation for a given Instrumented Library. Instrumented Library and Instrumentation Library may be the same Library if it has built-in OpenTelemetry instrumentation.
+
+There are two ways to instrument our app – manually and automatically:
+
+- **Auto instrumentation:** Automatically create spans from the application libraries we use with ready-to-use OpenTelemetry libraries.
+- **Manual instrumentation:** Manually add code to your application to define the beginning and end of each span and the payload.
+
+  To learn more, see [Instrumenting](/docs/concepts/instrumenting/).
+
+## Hello world: OpenTelemetry Go example
+
+We will start by creating our to-do service and installing two libraries (Gin and Mongo) to understand how instrumentations work.
+
+### Step 1: Create main.go file for our to-do app
+
+1. Install Gin and Mongo-driver
+
+```shell
+go get -u github.com/gin-gonic/gin
+go get go.mongodb.org/mongo-driver/mongo
+```
+
+2. Set up gin and mongo to listen on “/todo”
+
+3. Create some to-do’s to seed Mongo
+
+```go
+package main
+import (
+    "context"
+    "net/http"
+    "github.com/gin-gonic/gin"
+    "go.mongodb.org/mongo-driver/bson"
+    "go.mongodb.org/mongo-driver/mongo"
+    "go.mongodb.org/mongo-driver/mongo/options"
+)
+var client * mongo.Client
+func main() {
+    connectMongo()
+    setupWebServer()
+}
```

Comment:
Consider running a formatter over all the code code excerpts.
At a minimum, add a blank line between elements.